scripts/docker/debian/Dockerfile

```diff
@@ -0,0 +1,89 @@
+#
+# Copyright (c) .NET Foundation and contributors. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+#
+
+# Dockerfile that creates a container suitable to build dotnet-cli
+FROM debian:jessie
+
+# Misc Dependencies for build
+RUN apt-get update && apt-get -qqy install curl unzip gettext sudo
+
+# This could become a "microsoft/coreclr" image, since it just installs the dependencies for CoreCLR (and stdlib)
+RUN echo "deb http://llvm.org/apt/jessie/ llvm-toolchain-jessie-3.6 main"  | tee /etc/apt/sources.list.d/llvm.list && \
+    curl http://llvm.org/apt/llvm-snapshot.gpg.key | apt-key add - && \
+    apt-get update && apt-get -qqy install\
+	libc6 \
+	libedit2 \
+	libffi6 \
+	libgcc1 \
+	libicu52 \
+	liblldb-3.6 \
```

Comment:
> Jenkins - I wasn't sure the Jenkins integration was necessary for RC2 as I listed in my summary. Regardless can debian builds be added to Jenkins prior to merging this PR?

If we are generating debian tarballs regularly then we need Jenkins CI to make sure no one breaks it. It is just a oneline change at https://github.com/dotnet/cli/blob/rel/1.0.0/netci.groovy#L12 assuming the builds are started the same way as here - https://github.com/dotnet/cli/blob/rel/1.0.0/netci.groovy#L39. The change is to replace `Ubuntu:x64:Debug` with `Debian8.2:x64:Debug`. 

Jenkins will not picked it up immediately, but after merging the netci.groovy changes any new PR will include the Debian build check.